frontend/packages/knative-plugin/src/catalog/useEventSourceProvider.ts

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+import { CatalogItem, CatalogExtensionHook } from '@console/plugin-sdk';
+import { K8sKind, K8sResourceKind, referenceForModel } from '@console/internal/module/k8s';
+import {
+  useK8sWatchResource,
+  WatchK8sResource,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import { useEventSourceModels } from '../utils/fetch-dynamic-eventsources-utils';
+import { CamelKameletBindingModel, CamelKameletModel } from '../models';
+import { getEventSourceIcon } from '../utils/get-knative-icon';
+import { getEventSourceCatalogProviderData } from './event-source-data';
+
+const normalizeEventSources = (
+  eventSources: K8sKind[],
+  namespace: string,
+  t: TFunction,
+): CatalogItem[] => {
+  const normalizedEventSources: CatalogItem[] = eventSources.map((eventSource) => {
+    const { kind, label: name, id: uid } = eventSource;
+    const href = `/catalog/ns/${namespace}/eventsource?sourceKind=${kind}`;
+    const { documentationUrl, description, provider = 'Red Hat' } =
+      getEventSourceCatalogProviderData(kind) ?? {};
+    return {
+      uid,
+      name,
+      description,
+      documentationUrl,
+      icon: { url: getEventSourceIcon(referenceForModel(eventSource)), class: null },
+      type: 'EventSource',
+      provider,
+      cta: { label: t('knative-plugin~Create Event Source'), href },
+    };
+  });
+  return normalizedEventSources;
+};
+const normalizeKamelets = (
+  kamelets: K8sResourceKind[],
+  namespace: string,
+  t: TFunction,
+): CatalogItem[] => {
+  const normalizedKamelets = kamelets.map((k) => {
+    const {
+      kind,
+      metadata: { uid, name, creationTimestamp },
+      spec: {
+        definition: { title, description },
+      },
+    } = k;
+    const iconUrl = getEventSourceIcon(kind, k);
+    const href = `/catalog/ns/${namespace}/eventsource?sourceKind=${CamelKameletBindingModel.kind}&name=${name}`;
```

Comment:
Why provide `name` here but not in the other case?

Reply:
in case of `kamelets` name makes it unique so needed i think , unlike other sources https://github.com/openshift-integration/kamelet-catalog/blob/master/salesforce-source.kamelet.yaml#L2-L8